lib/features/steps/runner_steps.rb

```diff
@@ -62,3 +62,65 @@
   $logger.warn 'Sleep was used! Please avoid using sleep in tests!'
   sleep(seconds)
 end
+
+# Starts an interactive terminal
+When('I start a new terminal') do
+  Runner.stop_interactive_session
+  Runner.get_interactive_session
+end
+
+# Stops currently running interactive terminal
+When('I stop the current terminal') do
+  Runner.stop_interactive_session
+end
+
+# Run a command on the terminal
+#
+# @step_input command [String] The command to run on the terminal
+When('I input {string} interactively') do |command|
+  current_terminal = Runner.get_interactive_session
+  success = current_terminal.run_command(command)
+  assert(success, 'The terminal had already closed')
+end
+
+# Get the current buffer values in the terminal
+#
+# @step_input expected_chars [String] The chars present in current buffer
+Then('the terminal is outputting {string}') do |expected_chars|
+  current_terminal = Runner.get_interactive_session
+  assert_equal(expected_chars, current_terminal.current_buffer)
+end
+
+# Verify a string appears in the stdout logs
+#
+# @step_input expected_line [String] The string present in stdout logs
+Then('the terminal has output {string}') do |expected_line|
+  current_terminal = Runner.get_interactive_session
+  match = current_terminal.parsed_output.any? { |line| line == expected_line }
+  assert(match, "No output lines matched #{expected_line}")
+end
+
+# Verify a string appears in the stderr logs
+#
+# @step_input expected_err [String] The string present in stderr logs
+Then('the terminal has the error message {string}') do |expected_err|
+  current_terminal = Runner.get_interactive_session
+  match = current_terminal.parsed_errors.any? { |line| line == expected_err }
+  assert(match, "No output lines matched #{expected_err}")
+end
+
+# Verify the exit code of the terminal
+#
+# @step_input exit_code [Integer] The expected exit code
+Then('the terminal exit code equals {int}') do |exit_code|
```

Comment:
"terminal exit code" sounds to me like the terminal has finished and this is the exit code for the whole process.  But multiple commands can be run in the same terminal, right?  And this is just the exit code of the last command?  
Assuming I understand that correctly then maybe just reword slightly?
```suggestion
Then('the last terminal exit code is {int}') do |exit_code|
```
(Changed `equals` to `is` for more consistency, although there is probably a whole piece there for us across all the existing steps!).

Reply:
I don't believe we'll get an exit code through Open3 other than the shell exit code, so i believe this is valid.